### PR TITLE
Improve block processing

### PR DESCRIPTION
**Pull request by ebocher**
Now it uses graph technique to cluster the geometries.
It consumes less memory and it's faster than the previous method (100 X faster). 

About #114

### CODE REVIEW

File: geoindicators/src/main/groovy/org/orbisgis/SpatialUnits.groovy [hunk 1]
```diff
@@ -202,22 +205,63 @@ static IProcess prepareRSUData(){
...
-                datasource.execute "CREATE TABLE $outputTableName as  select  EXPLOD_ID as $columnIdName, the_geom "+
-                        "from st_explode ('(select ST_UNION(ST_ACCUM(ST_BUFFER(THE_GEOM,$distance))) as the_geom"+
-                        " from $inputTableName)')"
-                logger.info("The geometries have been merged")
+
+
+                //Find all neighbors for each building
+                logger.info("Building index to perform the process...")
+                if(!datasource.getSpatialTable(inputTableName).the_geom.indexed){
+                    datasource.getSpatialTable(inputTableName).the_geom.createSpatialIndex()
```
Comment (SPalominos) [inline geoindicators/src/main/groovy/org/orbisgis/SpatialUnits.groovy]: Indexes are created only if they not already exists so you don't need to test that before.

File: geoindicators/src/main/groovy/org/orbisgis/SpatialUnits.groovy [hunk 2]
```diff
@@ -202,22 +205,63 @@ static IProcess prepareRSUData(){
...
+                }
+                if(!datasource.getSpatialTable(inputTableName).id_build.indexed){
+                    datasource.getSpatialTable(inputTableName).id_build.createIndex()
```
Comment (SPalominos) [inline geoindicators/src/main/groovy/org/orbisgis/SpatialUnits.groovy]: Same thing

File: geoindicators/src/test/groovy/org/orbisgis/SpatialUnitsTests.groovy [hunk 3]
```diff
@@ -71,7 +71,7 @@ class SpatialUnitsTests {
         assertTrue blockP.execute([inputTableName: "build_tempo",distance:0.01,
                      prefixName: "block", datasource: h2GIS])
         def outputTable = blockP.results.outputTableName
-        def countRows = h2GIS.firstRow "select count(*) as numberOfRows from $outputTable"
+        def countRows = h2GIS.firstRow "select count(*) as numberOfRows from $outputTable".toString()
```
Comment (SPalominos) [inline geoindicators/src/test/groovy/org/orbisgis/SpatialUnitsTests.groovy]: `toString()` is not mandatory, you can remove it to simplify.

File: geoindicators/src/main/groovy/org/orbisgis/SpatialUnits.groovy (hunk 1 above)
Comment (ebocher) [reply geoindicators/src/main/groovy/org/orbisgis/SpatialUnits.groovy]: I will fix it

File: geoindicators/src/test/groovy/org/orbisgis/SpatialUnitsTests.groovy (hunk 3 above)
Comment (ebocher) [reply geoindicators/src/test/groovy/org/orbisgis/SpatialUnitsTests.groovy]: Ok

**Pull request merged**